src/client/shadows/dynamicshadows.cpp

```diff
@@ -96,11 +96,24 @@ DirectionalLight::DirectionalLight(const u32 shadowMapResolution,
 		farPlane(farValue), mapRes(shadowMapResolution), pos(position)
 {}
 
-void DirectionalLight::update_frustum(const Camera *cam, Client *client, bool force)
+void DirectionalLight::update_frustum(const Camera *cam, Client *client, bool force, float timeoftheday)
 {
 	if (dirty && !force)
 		return;
 
+	if (timeoftheday < 0.1 || timeoftheday > 0.9 || (timeoftheday > 0.4 && timeoftheday < 0.6)) {
+		/*
+		 * When we are close to noon or midnight shadows do not move much,
+		 * so shadow-jitter due to rounding is more visible.
+		 * At those times we update shadows less often.
+		 */
+		v3f q_direction  = quantizeDirection(direction, M_PI / 2880 /*15 seconds*/);
+		if (q_direction == last_quant_directin)
```

Comment:
float comparison is OK here, since quantizeDirection uses ceil for integer math.